CHANGELOG.md

```diff
@@ -7,6 +7,13 @@ and this project adheres to the versioning scheme outlined in the [README.md](RE
 
 ## [Unreleased]
 
+### Added
+
+- When the Clarity library is built with feature flag `developer-mode`, comments
+  from the source code are now attached to the `SymbolicExpression` nodes. This
+  will be useful for tools that use the Clarity library to analyze and
+  manipulate Clarity source code, e.g. a formatter.
```

Comment:
As implemented, I believe these changes are DB breaking changes (so it would require a resync from genesis), because the AST nodes will fail to parse database entries that existed before this patch. I think, if possible, we should avoid this, but if we can't, this needs to be stated in the CHANGELOG